### ISSUE TITLE
Validate/sanitize inputs more generally

### ISSUE THREAD

**Issue by thomased** [rmaia/pavo#102]
We've got some specific checks around, but might be nice to have some general checks & a generic error ('Input not recognised') that just catches all totally-wrong inputs. All the functions expect pretty specific stuff, so an internal checker function or two, that's then called near the beginning of main functions, would probably do it.

**Comment by Bisaloo** [comment]
I totally agree! Many functions use the same first lines of code and I would be great to refactor it to reduce code redundancy. I think the main challenge here is to stay flexible enough when checking the input. For example, I tend to work a lot with dataframes instead of dedicated S3 classes. There's nothing more annoying than functions that fail without explanations because you didn't provide the *exact* input they expect (matrices vs df, named rows, etc.).

**Comment by Bisaloo** [comment]
I'm wondering whether it would be worth adding a new dependency such as https://github.com/hadley/assertthat for this.
